bsip-0061.md

```diff
@@ -0,0 +1,79 @@
+
+# Motivation
+At present, the BitShares protocol does not allow direct modification of limit orders in the markets. Instead, to modify a limit order, the order must be canceled and and a new one created in its place. This results in greater-than-necessary overhead when updating a limit order, an action which is performed by market maker robots with great frequency (hundreds of times per day, per market). Additionally, the cancel/create combination requires payment of two fees, which has led to some pressure from market makers for BitShares to provide a cheaper mechanism for maintaining fresh order walls in the markets.
+
+# Rationale
+The specification below was chosen on the following criteria:
+ - Minimal storage space for historical data (i.e. serialized operation size)
+   - Cancel/create combination requires two distinct operations, and explicit restatement of all order parameters
+   - Update operation requires only one operation, which elides all fields which are not being updated
+ - Minimal processing
+   - No deletion/construction is necessary, object is modified in place
+   - Cancel/create requires two separate modifications of balances; update requires zero or one
+   - Order matching is only triggered if order is moved past previous top of book
+ - Minimal RAM consumption for undo storage
+   - Object modification requires slightly less space than object deletion and creation
+   - Balances and account statistics are not updated unless necessary, avoiding unnecessary undo storage
+
+# Specification
+A complete implementation of this BSIP is available for review [here](https://github.com/bitshares/bitshares-core/compare/develop...nathanhourt:issue-1604).
+
+The `limit_order_update_operation` contains the following fields:
+```cpp
+       asset fee;
+       account_id_type seller;
+       limit_order_id_type order;
+       optional<price> new_price;
+       optional<share_type> delta_amount_to_sell;
+       optional<time_point_sec> new_expiration;
+
+       extensions_type extensions;
+```
+
+This operation will have the following stateless validity checks:
+ - Fee is positive 
+ - At least one of the optional update fields is not null
+ - If `new_price` is not null, run its `validate` method
+ - If `delta_amount_to_sell` is not null, check it is not zero
+
+Additionally, the following stateful checks will be applied during evaluation:
+ - `seller` matches referenced order object's owner
+ - If `new_price` is not null, check its assets match those in the referenced order
+ - If `delta_amount_to_sell` is not null and positive, check it does not exceed seller's balance
+ - If `delta_amount_to_sell` is not null and negative, check it does not exceed order's balance
+ - Check updated order value is greater than zero
```

Comment:
Technically, we should do a `cull_small` check. Question: is it intended that this operation can be used as a cheaper alternative to cancel existing orders? What does "order value" mean?

Reply:
I utilized the same logic as the existing `maybe_cull_small_order` call and reject the transaction if it turns the order to dust.

Reply:
Revised the language to be clearer on this point